creation of install doc and refactor of doc in general (#1908)
ROCm/composable_kernel: docs/install/Composable-Kernel-Docker.rst

.. meta::
  :description: Composable Kernel docker files
  :keywords: composable kernel, CK, ROCm, API, docker

.. _docker-hub:

********************************************************************
Composable Kernel Docker containers
********************************************************************

Docker images that include all the required prerequisites for building Composable Kernel are available on `Docker Hub <https://hub.docker.com/r/rocm/composable_kernel/tags>`_.

> The images also contain `ROCm <https://rocm.docs.amd.com/en/latest/index.html>`_, `CMake <https://cmake.org/getting-started/>`_, and the `ROCm LLVM compiler infrastructure <https://rocm.docs.amd.com/projects/llvm-project/en/latest/index.html>`_.

Reviewer: contains or includes?

Author: The images are containers, so I went with contain. idk.

> Composable Kernel Docker images are named according to their operating system and ROCm version. For example, a Docker image named ``ck_ub22.04_rocm6.3`` would be an Ubuntu 22.04 image with ROCm 6.3. 

Reviewer: ```suggestion
Composable Kernel Docker images are named according to their operating system and ROCm version. For example, a Docker image named ``ck_ub22.04_rocm6.3`` represents Ubuntu 22.04 with ROCm 6.3. 
```

Author: I'll switch it to "is".

Author: > Composable Kernel Docker images are named according to their operating system and ROCm version. For example, a Docker image named ``ck_ub22.04_rocm6.3`` would correspond to an Ubuntu 22.04 image with ROCm 6.3.
